### PR TITLE
simplify artifacts download and some other things

### PR DESCRIPTION
**Pull request by martadinata666**
* Use nitnelave/rust-dev:latest for all job
* simplify artifacts download, no more per arch downloading
* add `rustup target add wasm32-unknown-unknown` on ui build, will this affect anything?
* update actions/cache version
* modify release to all in one compressed

### CODE REVIEW

File: .github/workflows/docker-build-static.yml [hunk 1]
```diff
@@ -100,7 +89,8 @@ jobs:
   build-armhf:
     runs-on: ubuntu-latest
     container:
-      image: rust:1.65
+#      image: rust:1.65
```
Comment (nitnelave) [inline .github/workflows/docker-build-static.yml:92]: Remove the comment.

File: .github/workflows/docker-build-static.yml [hunk 2]
```diff
@@ -481,12 +401,12 @@ jobs:
         id: create_release
         with:
           allowUpdates: true
-          artifacts:  bin/armhf-bin/lldap-armhf,
-                      bin/aarch64-bin/lldap-aarch64,
-                      bin/amd64-bin/lldap-amd64,
-                      bin/armhf-bin/migration-tool-armhf,
-                      bin/aarch64-bin/migration-tool-aarch64,
-                      bin/amd64-bin/migration-tool-amd64,
+          artifacts:  bin/aarch64-lldap,
```
Comment (nitnelave) [inline .github/workflows/docker-build-static.yml:404]: Should we maybe wrap them in a tar with the web assets as well?

Anyone who downloads the binary needs the web assets as well, and people who need just the web assets can then replace the binary with their own.

File: .github/workflows/docker-build-static.yml [hunk 3]
```diff
@@ -392,21 +392,41 @@ jobs:
...
+             cp -r app armhf-lldap/
+             ls -alR aarch64-lldap/
+             ls -alR amd64-lldap/
+             ls -alR armhf-lldap/
+
+      - name: Compress
+        run: |
+             sudo apt update
+             sudo apt install -y zip
+             zip -r aarch64-lldap.zip aarch64-lldap/
```
Comment (nitnelave) [inline .github/workflows/docker-build-static.yml:418]: tar?

Comment (nitnelave) [reply .github/workflows/docker-build-static.yml:418]: It's more standard for linux packages. zip for windows why not, but we don't have a windows binary.

Comment (nitnelave) [reply .github/workflows/docker-build-static.yml:418]: https://xkcd.com/1168/

File: .github/workflows/docker-build-static.yml [hunk 4]
```diff
@@ -413,20 +414,18 @@ jobs:
 
       - name: Compress
         run: |
-             sudo apt update
-             sudo apt install -y zip
-             zip -r aarch64-lldap.zip aarch64-lldap/
-             zip -r amd64-lldap.zip amd64-lldap/
-             zip -r armhf-lldap.zip armhf-lldap/
+             tar -czvf -r aarch64-lldap.tar.gz aarch64-lldap/
+             tar -czvf amd64-lldap.tar.gz amd64-lldap/
```
Comment (nitnelave) [inline .github/workflows/docker-build-static.yml:418]: I'm confused: do we need the `-r` or not?

Comment (nitnelave) [reply .github/workflows/docker-build-static.yml:418]: Yes, I think we want it for all.

Comment (nitnelave) [reply .github/workflows/docker-build-static.yml:418]: ah no you're creating a file that doesn't exist. Carry on! :)

**Comment by martadinata666** [comment]
The last part is verify docker image, it should be work, the artifacts path sometimes tricky 🤞🏼

**Comment by nitnelave** [comment]
> Release should be like this now https://github.com/martadinata666/lldap/releases/tag/v0.4.2-alpha8

The artifacts structure looks good!

Thanks a lot for your help with docker, this is not something that I enjoy :D

**Comment by nitnelave** [comment]
Just checking, are you checking the docker image?

**Comment by martadinata666** [comment]
No, not really, from the actions log, the structure should be fine. as far as I can tell. it had everything in `/app`

**Pull request merged**